lib/modules/ipfs/embarkjs.js

```diff
@@ -144,6 +134,10 @@ __embarkIPFS.register = function(addr, callback) {
     return new Error(NoConnectionError);
   }
 
+  if (addr.length !== 46 || !addr.startsWith('Qm')) {
```

Comment:
Are we sure about the startsWith Qm ?
It looks like it is correct but I could not find something that specify it.

Also I believe that if you map your dns to your ipns hash, it works too
e.g: ipfs.io

Reply:
I took it from here, the guy looks like he knows what he's talking about: https://ethereum.stackexchange.com/questions/14023/validating-ipfs-hash-string#comment15686_14024

I don't understand the dns thing. You can put `ipfs.io` inside the register?

Reply:
This is not the register but resolver function.
It is possible once you have register your content and got the ipns to do a DNS mapping and the resolver accept this behaviour
see the result of the command: `ipfs name --help`
```
  Resolve the value of a dnslink:
  
    > ipfs name resolve ipfs.io
    /ipfs/QmaBvfZooxWkrv7D3r8LS9moNjzD2o525XMZze69hhoxf5
```

Does it makes sense?

Reply:
Ok, I understand for the resolve. But does that affect the register? Because this condition is only for register

Reply:
Github tricked me by collapsing some changes. ignore my comments

Reply:
Hahahaha. I mean, I learned a bit about resolve still :)